packages/design-system-react-native/src/components/BadgeStatus/BadgeStatus.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+import React from 'react';
+import { View } from 'react-native';
+
+import {
+  DEFAULT_BADGESTATUS_PROPS,
+  TWCLASSMAP_BADGESTATUS_STATUS_BACKGROUNDCOLOR,
+  TWCLASSMAP_BADGESTATUS_STATUS_INNER_BORDERCOLOR,
+} from './BadgeStatus.constants';
+import type { BadgeStatusProps } from './BadgeStatus.types';
+
+const BadgeStatus = ({
+  status,
+  size = DEFAULT_BADGESTATUS_PROPS.size,
+  hasBorder = DEFAULT_BADGESTATUS_PROPS.hasBorder,
+  twClassName = '',
+  style,
+  ...props
+}: BadgeStatusProps) => {
+  const tw = useTailwind();
+
+  return (
+    <View
+      style={[
+        tw`
+        self-start
+        rounded-full 
+        ${hasBorder ? 'border-[2px] border-background-default' : ''} 
```

Comment:
non-blocking: While there's no performance benefit in`twrnc` (all styles are processed into objects at runtime), consider using `border-2` instead of `border-[2px]` for cross-platform consistency. This would align with web Tailwind conventions where predefined classes do offer performance benefits, making our components more portable across platforms.
```suggestion
        ${hasBorder ? 'border-2 border-background-default' : ''} 
```